addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -67,6 +67,32 @@ function factory(dependencies) {
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         * @private
+         * @returns {mail.messaging}
+         */
+        _computeMessaging() {
+            return [['link', this.env.messaging]];
+        }
+
+        /**
+         * @private
+         * @returns {String[]}
+         */
+        _computeTextInputSendShortcuts() {
+            if (!this.thread) {
+                return;
+            }
+            if (
+                this.env.messaging.device.isMobile ||  // The goal here is to target small screen size
+                (this.thread.model === 'mail.channel' &&
+                this.thread.mass_mailing)  // Allow to send the messages for mailing channel with shortcut
```

Comment:
I don't understand this comment especially the "with shortcut" part?

Also the guideline for multi line conditions is one per line, and the `(` and `)` each on their own lines too.
But in this case it's probably better to keep those 2 in a single line, or create a variable for them (such as `isMailingList`) to avoid mixing `and` and `or` inside the `if`.